### ISSUE TITLE
Fail create hdf5.File - signature assert

### ISSUE THREAD

**Issue by Rich-Taylor-oss** [usnistgov/jsfive#20]
I am trying to load an HDF5 file in node v14.17.5 and getting an error when creating a new hdf5.File - `new hdf5.File(ab.buffer);`

Sample file: https://water.weather.gov/precip/downloads/current/nws_precip_1hour_conus.nc

As far as I can tell in my debugging, it is processing the very start of the file.  It's (erroneously?) expecting to see a 'FRHP' signature, but the signature is the standard HDF signature that should be at the start of the file.

As a side note, the assert function does not handle false very well, trying to call it as a function.

```
home/rtaylor/src/jsfive/dist/index.js:156
    thing();
    ^

TypeError: thing is not a function
    at assert (/home/rtaylor/src/jsfive/dist/index.js:156:5)
    at new FractalHeap (/home/rtaylor/src/jsfive/dist/index.js:9424:22)
    at DataObjects._iter_links_btree_v2 (/home/rtaylor/src/jsfive/dist/index.js:10331:16)
    at _iter_links_btree_v2.next (<anonymous>)
    at DataObjects._iter_link_from_link_info_msg (/home/rtaylor/src/jsfive/dist/index.js:10325:19)
    at _iter_link_from_link_info_msg.next (<anonymous>)
    at DataObjects.iter_links (/home/rtaylor/src/jsfive/dist/index.js:10224:21)
    at iter_links.next (<anonymous>)
    at Function.fromEntries (<anonymous>)
    at DataObjects.get_links (/home/rtaylor/src/jsfive/dist/index.js:10214:19)```

**Comment by bmaranville** [comment]
Can you try the version I just uploaded (v0.3.4)?  I found the bug, I think.  I had never tested this particular path before - where link_info_msg is present but the links are stored compactly (so the address is stored as UNDEFINED_ADDRESS).  I had converted undefined address to null in the first check, so the second check for UNDEFINED_ADDRESS was failing.

If it works for you I can publish to npm.

**Comment by bmaranville** [comment]
The assert function is kind of a cheat - a quick way to throw an error for anything if it gets called.

**Comment by bmaranville** [comment]
Ah no - it doesn't fix the problem... just reveals another issue.  Might take me a little bit longer to fix.

**Comment by bmaranville** [comment]
Ok, it should be fixed with the latest commit d2bc3558fd32a10afa5e764e48897ae1f734c8b3 ... I was able to fully load and browse your dataset with this change.

**Comment by Rich-Taylor-oss** [comment]
Yes, I'm able to load and view the dataset now.  Thanks for the quick response!
